tests/device_tests/test_msg_backup_device.py

```diff
@@ -111,6 +112,37 @@ def test_backup_slip39_advanced(client: Client, click_info: bool):
     assert expected_ms == actual_ms
 
 
+@pytest.mark.skip_t1b1
+@pytest.mark.setup_client(needs_backup=True, mnemonic=MNEMONIC_SLIP39_ADVANCED_20)
+@pytest.mark.parametrize(
+    "share_threshold,share_count",
+    [(1, 1), (2, 2), (3, 5)],
+    ids=["1_of_1", "2_of_2", "3_of_5"],
+)
+def test_backup_slip39_custom(client: Client, share_threshold, share_count):
+    assert client.features.needs_backup is True
+
+    with client:
+        IF = InputFlowSlip39CustomBackup(client, share_count)
+        client.set_input_flow(IF.get())
+        device.backup(
+            client, group_threshold=1, groups=[(share_threshold, share_count)]
+        )
+
+    client.init_device()
+    assert client.features.initialized is True
+    assert client.features.needs_backup is False
+    assert client.features.unfinished_backup is False
+    assert client.features.no_backup is False
+    assert client.features.backup_type is messages.BackupType.Slip39_Advanced
+
+    expected_ms = shamir.combine_mnemonics(MNEMONIC_SLIP39_ADVANCED_20)
+    actual_ms = shamir.combine_mnemonics(
+        IF.mnemonics[:3] + IF.mnemonics[5:8] + IF.mnemonics[10:13]
+    )
+    assert expected_ms == actual_ms
```

Comment:
you should probably be using `MNEMONIC_SLIP39_BASIC_20_3of6` because for that you also get a `MNEMONIC_SLIP39_BASIC_20_3of6_SECRET` to test against, instead of having to call `combine_mnemonics`

the code for `actual_ms` is wrong
far as I can understand, `IF.mnemonics` is a list of individual shares, right?
you should `assert len(IF.mnemonics) == share_count`
you should also assert that any(?) combination of `share_threshold` mnemonics out of `IF.mnemonics` combines into `MNEMONIC_SLIP39_BASIC_20_3of6_SECRET`

(honestly, probably just checking the last N shares is enough, no need to iterate the combinations)

Reply:
39e10c48c0c2978f199d4847a423bf14b36fa8e3